airflow/providers/elasticsearch/log/es_task_handler.py

```diff
@@ -152,7 +153,8 @@ def __init__(
         offset_field: str = "offset",
         host: str = "http://localhost:9200",
         frontend: str = "localhost:5601",
-        index_patterns: str | None = conf.get("elasticsearch", "index_patterns", fallback="_all"),
+        index_patterns: str = conf.get("elasticsearch", "index_patterns"),
+        index_patterns_callable: str = conf.get("elasticsearch", "index_patterns_callable", fallback=""),
```

Comment:
Shouldn't the type here be `typing.Callable`?

Reply:
Maybe we should add these params in the doc string.

Reply:
historically if a param comes from the config file then it is not added in doc string you can see `index_patterns`

Reply:
IMO, If it always comes from a config file, we can remove it as a param? Otherwise, devs can still use and create the class object passing it. I see no harm in adding it to docs. 

But TBH I'm not sure what the norm is here :)

Reply:
If want to add I'll add all of them in a separate PR